integration-cli/docker_cli_exec_test.go

```diff
@@ -616,9 +616,10 @@ func (s *DockerSuite) TestExecStartFails(c *check.C) {
 	testRequires(c, DaemonIsLinux)
 	name := "exec-15750"
 	dockerCmd(c, "run", "-d", "--name", name, "busybox", "top")
+	c.Assert(waitRun(name), check.IsNil)
 
-	_, errmsg, status := dockerCmdWithStdoutStderr(nil, "exec", name, "no-such-cmd")
-	if status == 255 && !strings.Contains(errmsg, "executable file not found") {
-		c.Fatal("exec error message not received. The daemon might had crashed")
+	out, status, _ := dockerCmdWithError("exec", name, "no-such-cmd")
+	if status == 255 && !strings.Contains(out, "executable file not found") {
```

Comment:
This check is a bit odd. If status != 255 (let's say its zero), the 'if' will fail and the test will pass.  Seems like it should be:
```
if status != 255 || !string.Contains(...) {
```
no?